src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#ifndef TRINITY_CHASEMOVEMENTGENERATOR_H
+#define TRINITY_CHASEMOVEMENTGENERATOR_H
+
+#include "MovementGenerator.h"
+#include "Unit.h"
+#include "Optional.h"
+#include "PathGenerator.h"
+
+class ChaseMovementGenerator : public MovementGenerator, public AbstractFollower
+{
+    public:
+        MovementGeneratorType GetMovementGeneratorType() const override { return CHASE_MOTION_TYPE; }
+
+        ChaseMovementGenerator(Unit* target, Optional<ChaseRange> range = {}, Optional<ChaseAngle> angle = {}) : AbstractFollower(ASSERT_NOTNULL(target)), _range(range), _angle(angle) {}
+
+        void Initialize(Unit* owner) override { owner->AddUnitState(UNIT_STATE_CHASE); owner->SetWalk(false); }
```

Comment:
Why is walk/run mode tied to chasing? (http://www.wowhead.com/npc=20064/thaladred-the-darkener)